secure/validator.go

```diff
@@ -105,7 +108,31 @@ func (v JWSValidator) Validate(token *Token) (valid bool, err error) {
 	if err != nil {
 		return
 	}
-
+	
+	if caps, ok := jwsToken.Payload().(jws.Claims).Get("capabilities").([]string); ok && len(caps) > 0 {
+		var valid_capabilities bool
+		for _, cap := range caps {
+			pieces := strings.Split(cap, ":")
+			
+			if len(pieces) == 5     &&
+			   pieces[0] == "x1"    && 
+			   pieces[1] == "webpa" && 
+			  (pieces[4] == "all" || pieces[4] == ctx.Value("method")) {
+				
+				claimPath := fmt.Sprintf("/%s/[^/]+/%s", pieces[2],pieces[3])
+				match_capPath, _ := regexp.MatchString(claimPath, ctx.Value("path").(string))
```

Comment:
Instead of regular expressions, why not use file globbing?  It's simpler and much cheaper to execute.

https://godoc.org/path/filepath#Match